### PR TITLE
chore: update Jina Embedder usage for V3 release

### PR DESCRIPTION
**Pull request by DresAaron**

### CODE REVIEW

**Comment by DresAaron** [comment]
@JoanFM could you review the changes please?

File: integrations/jina/tests/test_document_embedder.py
Comment (JoanFM) [inline integrations/jina/tests/test_document_embedder.py:250]: how are we sure that the task-type makes it into the request?

File: integrations/jina/tests/test_document_embedder.py
Comment (JoanFM) [inline integrations/jina/tests/test_document_embedder.py:299]: I do not see this tests anything. Also, this requirement is false in the API

**Review by JoanFM — APPROVED** [review]

**Comment by silvanocerza** [comment]
@DresAaron can you please sign the CLA? Can't merge this otherwise. :)

File: integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py
Comment (silvanocerza) [inline integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py:114]: If those are an empty string or `0` you'd miss the info.

```suggestion
        if self.task is not None:
            kwargs["task"] = self.task
        if self.dimensions is not None:
            kwargs["dimensions"] = self.dimensions
```

File: integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py
Comment (silvanocerza) [inline integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py:205]: Same as above.

```suggestion
        if self.task is not None:
            parameters["task"] = self.task
        if self.dimensions is not None:
            parameters["dimensions"] = self.dimensions
```

File: integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py
Comment (silvanocerza) [inline integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py:163]: This would be more concise, but either way works. Your call.

```suggestion
                json={"input": batch, "model": self.model_name, **(parameters or {})},
```

File: integrations/jina/src/haystack_integrations/components/embedders/jina/text_embedder.py
Comment (silvanocerza) [inline integrations/jina/src/haystack_integrations/components/embedders/jina/text_embedder.py:97]:
```suggestion
        if self.task is not None:
            kwargs["task"] = self.task
        if self.dimensions is not None:
            kwargs["dimensions"] = self.dimensions
```

**Review by silvanocerza — APPROVED** [review]
Nice!

**Comment by silvanocerza** [comment]
I'll go forward with a new release. :)

**Pull request merged**
